src/otp/base32.ts

```diff
@@ -0,0 +1,150 @@
+/*!
+ * base-32.js
+ * Copyright(c) 2024 Reaper
+ * MIT Licensed
+ */
+
+// Simple implementation based of RFC 4648 for base32 encoding and decoding
+
+const pad = "=";
+const base32alphaMap = {
```

Comment:
We might try to compress it using an array.

Reply:
doable, though the reference and number of operation would remain similar during `encode`, saves a bit more during `decode` so I've pushed the changes